### PR TITLE
add xunit logger provider

### LINKED ISSUE #243: Create "xunit" logger provider
In xunit, it is recommended that "console" output be written to `ITestOutputHelper` rather than the Console. We should provide a Logger to plug in to `ILoggerFactory` which writes to this. See https://github.com/aspnet/MusicStore/blob/dev/test/E2ETests/Common/XunitLogger.cs for an example.

/cc @natemcmaster

### PR DESCRIPTION
**Pull request by anurse**
fixes #243 

I was getting ready to use this in yet another SignalR test project and was like "I should put this somewhere more common" :).

/cc @Eilon @moozzyk

### CODE REVIEW

File: src/Microsoft.AspNetCore.Testing/xunit/XUnitLoggerFactoryExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Testing.xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Extensions.Logging
+{
+    public static class XUnitLoggerFactoryExtensions
+    {
+        public static void AddXUnit(this ILoggerFactory self, ITestOutputHelper output)
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Testing/xunit/XUnitLoggerFactoryExtensions.cs]: Nit: self => loggerFactory

File: src/Microsoft.AspNetCore.Testing/xunit/XUnitLoggerProvider.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+using System;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Testing.xunit
+{
+    public class XUnitLoggerProvider : ILoggerProvider
+    {
+        private ITestOutputHelper _output;
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Testing/xunit/XUnitLoggerProvider.cs]: Nit: readonly

**Comment by anurse** [comment]
Hrm... I guess we'll have to do something different then. Maybe just put it in Common... Stupid circular dependencies :)

**Comment by natemcmaster** [comment]
How about https://github.com/aspnet/Logging/blob/dev/src/Microsoft.Extensions.Logging.Testing?

**Comment by anurse** [comment]
Now that just makes perfect sense.

**Comment by anurse** [comment]
Opened aspnet/Logging#575 for that.

**Pull request closed without merging**